packages/runtime-dom/src/helpers/useCssVars.ts

```diff
@@ -62,10 +71,26 @@ function setVarsOnVNode(vnode: VNode, vars: Record<string, string>) {
     vnode = vnode.component.subTree
   }
 
-  if (vnode.shapeFlag & ShapeFlags.ELEMENT && vnode.el) {
-    setVarsOnNode(vnode.el as Node, vars)
+  if (vnode.shapeFlag & ShapeFlags.ELEMENT) {
+    if (!flag && vnode.el) {
+      setVarsOnNode(vnode.el as Node, vars)
+    }
+    if (isArray(vnode.children)) {
+      ;(vnode.children as VNode[]).forEach(c =>
+        setVarsOnVNode(c, vars, createObserve, true)
+      )
+    }
   } else if (vnode.type === Fragment) {
-    ;(vnode.children as VNode[]).forEach(c => setVarsOnVNode(c, vars))
+    ;(vnode.children as VNode[]).forEach(c =>
+      setVarsOnVNode(c, vars, createObserve, flag)
+    )
+  } else if (vnode.type === Teleport) {
+    if (vnode.target) {
+      ;(vnode.children as VNode[]).forEach(c =>
+        setVarsOnVNode(c, vars, createObserve)
+      )
+      createObserve(vnode.target as Node)
```

Comment:
This is not the correct fix. The `target` of `Teleport` is often the `body`, and if we observe the `body`, the cost will be enormous. Therefore, this solution is not reasonable.